src/analytics/api/models/base/data_processing.py

```diff
@@ -89,6 +90,16 @@ def air_quality_data():
 
                 }
             }
+            def replace_nan_with_null(obj):
```

Comment:
Thanks @wabinyai .....already merged but leaving a performance optimisation consideration.....

The function uses list comprehensions and dictionary comprehensions for lists and dictionaries, which is generally efficient. However, for very large nested structures, the recursive nature of the function could lead to a stack overflow error due to Python's recursion limit. This is unlikely to be an issue for most practical use cases but is something to be aware of.

Reply:
Thanks .

Reply:
Noted with thanks.

Reply:
@wabinyai  , additionally...
The function correctly handles lists and dictionaries by recursively calling itself on their elements. However, it does not explicitly handle other data types that might contain NaN values, such as numpy arrays or pandas DataFrames. If the input object contains such data types, the function will not replace NaN values within them.